src/cdk/explorer/nodes/constructNode.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { AWSTreeNodeBase } from '../../../shared/treeview/nodes/awsTreeNodeBase'
+import { cdk } from '../../globals'
+import { CfnResourceKeys, ConstructTreeEntity } from '../tree/types'
+
+/**
+ * Represents a CDK construct
+ */
+export class ConstructNode extends AWSTreeNodeBase {
+    private readonly type: string
+
+    get tooltip(): string {
+        return this.type || this.construct.path
+    }
+
+    get id(): string {
+        return this.construct.path
+    }
+
+    get iconPath(): string | vscode.Uri | { light: string | vscode.Uri; dark: string | vscode.Uri } | vscode.ThemeIcon {
+        if (this.type) {
+            return {
+                light: cdk.iconPaths.light.cloudFormation,
+                dark: cdk.iconPaths.dark.cloudFormation
+            }
+        }
+
+        return {
+            light: cdk.iconPaths.light.cdk,
+            dark: cdk.iconPaths.dark.cdk
+        }
+    }
+
+    public constructor(
+        public readonly label: string,
+        public readonly collapsibleState: vscode.TreeItemCollapsibleState,
+        public readonly construct: ConstructTreeEntity
+    ) {
+        super(construct.id, collapsibleState)
+        this.contextValue = 'awsCdkNode'
+
+        this.type = construct.attributes ? (construct.attributes[CfnResourceKeys.TYPE] as string) : ''
```

Comment:
This pattern is repeated. Consider a "getTypeAttributeOrDefault(construct, default)" utility method.

Reply:
Is `construct.attributes[CfnResourceKeys.TYPE]` guaranteed if you have a `construct.attributes`? You may want to make the attributes into an interface or assert its existence instead of casting

Reply:
@bryceitoc9 - it's not guaranteed... the `attributes` section of a construct tree is essentially a bag of any number / type of properties. For resources that represent CloudFormation resources, it will include the CloudFormation resource type (i.e. `AWS::SNS::Topic` under this key by convention

Reply:
I think this was misinterpreted: the code suggests that at this point, if you have a construct.attributes, you'll always have an attribute `CfnResourceKeys.TYPE`? 

If that's the case, would it be worth having an interface for required attributes that you can set and pull from?

Reply:
you're right, will fix not to assume that the key will exist. If it has attributes, it might have a `type` but it's not going to be guaranteed.